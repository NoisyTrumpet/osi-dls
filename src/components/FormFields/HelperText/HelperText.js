import React from 'react';
import PropTypes from 'prop-types';
import './HelperText.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const HelperText = ({ addClass, error, children }) => {
  const classes = classNames(
    'helper-text',
    addClass,
    { error },
  );

  return (
    <div className={classes} role={error ? 'alert' : undefined}>
      <Typography variant="helper">
        {children}
      </Typography>
    </div>
  );
};

HelperText.defaultProps = {};

HelperText.propTypes = {
  addClass: PropTypes.string,
  error: PropTypes.bool,
  children: PropTypes.node,
};

export default HelperText;
