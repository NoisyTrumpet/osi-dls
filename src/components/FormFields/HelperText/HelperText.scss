.helper-text {
  margin-top: 6px;
  padding-left: 10px;

  &.error .typography-helper {
    color: map-get($colors, 'util-error');
  }
}

