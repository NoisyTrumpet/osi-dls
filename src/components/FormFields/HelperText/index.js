export { default as HelperText } from './HelperText'; // eslint-disable-line import/prefer-default-export
