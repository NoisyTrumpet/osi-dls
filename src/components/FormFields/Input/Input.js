import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss';
import classNames from 'classnames';

const Input = ({
  ariaControls,
  autoComplete,
  click,
  domId,
  error,
  isChecked,
  isDisabled,
  maxLength,
  name,
  onBlur,
  onChange,
  onFocus,
  placeholder,
  required,
  type,
  value,
  // range input
  min,
  max,
  step,
}) => {
  const classes = classNames(
    'input-field',
    { error },
  );

  return (
    <input
      aria-controls={ariaControls}
      autoComplete={autoComplete || undefined}
      type={type}
      name={name}
      id={domId}
      className={classes}
      maxLength={maxLength}
      placeholder={placeholder}
      defaultValue={value}
      checked={isChecked}
      disabled={isDisabled}
      aria-required={required}
      aria-invalid={error}
      onChange={onChange}
      onBlur={onBlur}
      onClick={click}
      onFocus={onFocus}
      min={min}
      max={max}
      step={step}
    />
  );
};

Input.defaultProps = {
  domId: '',
  name: '',
  onChange: null,
  onBlur: null,
  placeholder: '',
  type: 'text',
  value: '',
};

Input.propTypes = {
  ariaControls: PropTypes.string,
  autoComplete: PropTypes.oneOf(['on', 'off']),
  click: PropTypes.func,
  domId: PropTypes.string,
  error: PropTypes.bool,
  isChecked: PropTypes.bool,
  isDisabled: PropTypes.bool,
  maxLength: PropTypes.number,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  min: PropTypes.string,
  max: PropTypes.string,
  step: PropTypes.string,
};

export default Input;
