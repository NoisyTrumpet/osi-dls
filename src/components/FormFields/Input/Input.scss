.input-field {
  color: map-get($colors, 'brand-grey');
  border: rem(2px) solid map-get($colors, 'gray3');
  border-radius: $space-8;
  order: 1;
  padding: $space-12;
  @include fontSans($beam-84, $beam-vf, 'sans-serif', 84, 700);

  &:focus {
    border-color: map-get($colors, 'brand-blue');
    outline: none;
  }

  &[type='checkbox'],
  &[type='radio'] {
    cursor: pointer;
    opacity: 0;
    position: absolute;
    z-index: 1;
    @include appearance(none);
  }

  &[type='checkbox'],
  &[type='radio'] {
    height: rem(18px);
    width: rem(18px);
  }

  &[type='text'] {
    width: 100%;
  }

  &:disabled {
    color: map-get($colors, 'disabled');
    cursor: default;
    border-color: map-get($colors, 'disabled');

    ~ .input-label {
      cursor: default;

      .typography {
        color: map-get($colors, 'disabled');
      }
    }

    ~ .indicator {
      border-color: map-get($colors, 'disabled');

      &:after {
        background-color: map-get($colors, 'disabled');
      }
    }
  }

  &.error {
    border-color: map-get($colors, 'util-error');

    &:focus {
      border-color: map-get($colors, 'util-error');
    }
  }
}

.input-field-text {
  .helper-text {
    padding-left: 15px;
  }
}

.input-field-select {
  .helper-text {
    padding-left: 15px;
  }
}

// container ... normally goes in a forms.scss
.input-field-checkbox,
.input-field-radio {
  align-items: center;
  display: flex;
  position: relative;

  .input-field,
  .input-label {
    &:hover {
      ~ .indicator {
        background-color: map-get($colors, 'gray1');
      }
    }
  }

  .input-field {
    &:checked {
      ~ .indicator {
        background-color: map-get($colors, 'brand-blue');

        &:after {
          opacity: 1;
        }
      }
    }

    &:disabled {
      ~ .indicator {
        background-color: map-get($colors, 'brand-transparent');
      }
    }
  }

  .input-label {
    cursor: pointer;
    line-height: $space-24;
    @include user-select;

    &.isDisabled {
      cursor: auto;
    }
  }
}

.input-field-checkbox {
  .helper-text {
    margin: 0 auto 6px -26px;
  }
}

// Prevent visual toggle when helpertext appears
.no-helpertext {
  margin-bottom: 30px;
}
