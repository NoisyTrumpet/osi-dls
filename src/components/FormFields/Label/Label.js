import React from 'react';
import PropTypes from 'prop-types';
import './Label.scss';
import classNames from 'classnames';
import { Typography } from '../../Typography';

const Label = ({
  fieldType,
  forId,
  display,
  hideLabel,
  isDisabled,
  color,
  required,
  addClass,
  suffixHelperText,
  children,
}) => {
  const classes = classNames(
    'input-label',
    fieldType,
    display,
    hideLabel && 'hide-label',
    addClass,
    { isDisabled },
  );

  const srtVariant = hideLabel && "srt";

  return (
    <label
      className={classes}
      htmlFor={forId}
    >
      <Typography variant={srtVariant || "field-label"} color={color} addClass={addClass}>
        {children}
      </Typography>
      {suffixHelperText && (
        <Typography variant="helper" addClass="optional">
          {suffixHelperText}
        </Typography>
      )}
      {!required && (
        <Typography variant={srtVariant || "helper"} addClass="optional">
          (Optional)
        </Typography>
      )}
    </label>
  );
};

Label.defaultProps = {
  fieldType: 'text',
  display: 'inline',
  isDisabled: false,
  color: '',
};

Label.propTypes = {
  fieldType: PropTypes.oneOf([
    'text',
    'radio',
    'checkbox',
    'textarea',
    'select',
  ]),
  forId: PropTypes.string,
  display: PropTypes.oneOf([
    'inline',
    'inline-block',
    'block',
  ]),
  hideLabel: PropTypes.bool,
  isDisabled: PropTypes.bool,
  color: PropTypes.string,
  suffixHelperText: PropTypes.string, // helper text that is appended to label
  required: PropTypes.bool,
  addClass: PropTypes.string,
  children: PropTypes.node,
};

export default Label;
