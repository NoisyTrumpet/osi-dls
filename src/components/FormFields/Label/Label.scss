.input-label {
  color: map-get($colors, 'gray5');
  padding-left: $space-8;
  order: 2;

  &.block {
    display: block;
  }

  &.inline-block {
    display: inline-block;
  }

  &.text {
    position: absolute;
    top: rem(-11px);
    left: rem(2px);
    transform: translate(0, 50%);

    .optional {
      font-size: rem(12px);
      padding-left: rem(2px);
      padding-right: 0px;
      position: absolute;
      background: #ffffff;
    }
    
    .typography-field-label {
      background: map-get($colors, 'brand-white');
      padding: 0 $space-4;
    }
  }
}
