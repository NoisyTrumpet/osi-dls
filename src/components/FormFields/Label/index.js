export { default as Label } from './Label'; // eslint-disable-line import/prefer-default-export
