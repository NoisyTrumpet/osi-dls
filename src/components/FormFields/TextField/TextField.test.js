import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Formik, Form } from 'formik';
import { Input } from '../Input';
import { TextField } from './';
import { Label } from '../Label';
import { HelperText } from '../HelperText';

describe('<TextField />', () => {
  it('Should render', () => {
    const wrapper = mount(
      <Formik initialValues={{ textField: '' }}>
        <Form>
          <TextField
            label="label"
            name="textField"
            domId="textFieldId"
            helperText="Optional helper text"
          />
        </Form>
      </Formik>,
    );
    expect(wrapper.find(Label).length).toBe(1);
    expect(wrapper.find(HelperText).length).toBe(1);
    expect(wrapper.find(Input).length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
