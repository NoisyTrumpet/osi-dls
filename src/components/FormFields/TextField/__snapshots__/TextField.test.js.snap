// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Should render 1`] = `
<Formik
  initialValues={
    Object {
      "textField": "",
    }
  }
>
  <Form>
    <form
      action="#"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <TextField
        domId="textFieldId"
        helperText="Optional helper text"
        iconOnClick={null}
        label="label"
        name="textField"
      >
        <div
          className="input-field-container input-field-text"
          data-testid="test-id-input"
        >
          <Input
            domId="textFieldId"
            name="textField"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="text"
            value=""
          >
            <input
              className="input-field"
              defaultValue=""
              id="textFieldId"
              name="textField"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="text"
            />
          </Input>
          <Label
            color="gray4"
            display="block"
            fieldType="text"
            forId="textFieldId"
            isDisabled={false}
          >
            <label
              className="input-label text block"
              htmlFor="textFieldId"
            >
              <Typography
                color="gray4"
                variant="field-label"
              >
                <span
                  className="typography typography-field-label color-gray4"
                >
                  label
                </span>
              </Typography>
              <Typography
                addClass="optional"
                variant="helper"
              >
                <span
                  className="typography typography-helper optional"
                >
                  (Optional)
                </span>
              </Typography>
            </label>
          </Label>
          <HelperText>
            <div
              className="helper-text"
            >
              <Typography
                variant="helper"
              >
                <span
                  className="typography typography-helper"
                >
                  Optional helper text
                </span>
              </Typography>
            </div>
          </HelperText>
        </div>
      </TextField>
    </form>
  </Form>
</Formik>
`;
